kani-driver/src/assess.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::{collections::HashMap, path::PathBuf};
+
+use anyhow::Result;
+use comfy_table::Table;
+use kani_metadata::KaniMetadata;
+
+use crate::harness_runner::HarnessResult;
+use crate::session::KaniSession;
+
+/// Set some defaults for how we format tables
+fn assess_table_new() -> Table {
+    use comfy_table::*;
+
+    let mut table = Table::new();
+    table.set_content_arrangement(ContentArrangement::Dynamic);
+    table
+        .load_preset(comfy_table::presets::NOTHING)
+        .set_style(TableComponent::BottomBorder, '=')
+        .set_style(TableComponent::BottomBorderIntersections, '=')
+        .set_style(TableComponent::TopBorder, '=')
+        .set_style(TableComponent::TopBorderIntersections, '=')
+        .set_style(TableComponent::HeaderLines, '-')
+        .set_style(TableComponent::MiddleHeaderIntersections, '+')
+        .set_style(TableComponent::VerticalLines, '|');
+    table
+}
+
+fn unsupported_features_table(metadata: &KaniMetadata) -> Table {
+    // Map "unsupported feature name" -> (crates impacted, instance of use)
+    let mut counts: HashMap<String, (usize, usize)> = HashMap::new();
+
+    for item in &metadata.unsupported_features {
+        // key is unsupported feature name
+        let mut key = item.feature.clone();
+        // There are several "feature for <instance of use>" unsupported features.
+        // We aggregate those here by reducing it to just "feature"
+        if let Some((prefix, _)) = key.split_once(" for ") {
+            key = prefix.to_string();
+        }
+        let entry = counts.entry(key).or_default();
+        // crates impacted:
+        entry.0 += 1;
+        // instances of use:
+        entry.1 += item.locations.len();
+    }
+
+    // Sort descending by number of crates impacted by this missing feature
+    let mut sorted_counts: Vec<(String, (usize, usize))> = counts.into_iter().collect();
+    sorted_counts.sort_by_key(|i| usize::MAX - i.1.0);
```

Comment:
Why not just `|i| - i.1.0`

Reply:
It's an unsigned type. I could cast it or something, but this was easy. I did think it was a bit weird there wasn't a better way to reverse the sort order...